### PR TITLE
Remove jQuery -  Part 1

### PR DESCRIPTION
**Pull request by TorstenDittmann**
In this PR I replaced jQuery in following files:

- src/block.js 
- src/clipboard.js
- src/core.js 
- src/create-default-behavior.js
- src/cursor.js

and created a helper at `src/util/dom.js`

### CODE REVIEW

File: src/util/dom.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @param {HTMLElement | Array | String} target
+ * @param {Document} document
+ */
+export const domArray = (target, document) => {
+  let targets = []
+  switch (true) {
+    case isString(target):
+      targets = Array.from(document.querySelectorAll(target))
+      break
+    case target instanceof Array:
```
Comment (marcbachmann) [inline src/util/dom.js:13]: Can we use `Array.isArray(target)` here?
That makes the function more compatible for Array instances of other documents, e.g. an iframe.

File: src/core.js [hunk 2]
```diff
@@ -138,7 +138,7 @@ const Editable = module.exports = class Editable {
   disable (target) {
     const document = this.win.document
 
-    target = target ? [target] : Array.from(document.querySelectorAll(`.${config.editableClass}`))
+    target = target ? [target] : [...document.querySelectorAll(`.${config.editableClass}`)]
```
Comment (marcbachmann) [inline src/core.js:141]: We could also replace the `target.forEach` call with a `for (const element of target)` loop. That way this conversion won't be necessary at all.

File: src/util/dom.js (hunk 1 above)
Comment (TorstenDittmann) [reply src/util/dom.js:13]: Today I learned: `Array.isArray(target)` exists!

Didn't know that before 😄 Gonna adapt my code.

**Review by marcbachmann — APPROVED** [review]
Perfect. thank you

File: src/core.js [hunk 3]
```diff
@@ -138,19 +150,19 @@ const Editable = module.exports = class Editable {
...
-
+  enable (target, normalize) {
+    const document = this.win.document
     const shouldSpellcheck = this.config.browserSpellcheck
-    $elem.each((i, el) => {
-      block.init(el, {normalize, shouldSpellcheck})
-      this.dispatcher.notify('init', el)
-    })
+
+    target = target ? [target] : document.querySelectorAll(`.${config.editableDisabledClass}`)
```
Comment (peyerluk) [inline src/core.js:161]: We could use a small helper here which would also keep this working with jquery elements.
```js
function toElementArray (node) {
  return node.jquery ? node.toArray() : [node]
}
```

Comment (TorstenDittmann) [reply src/core.js:161]: Sounds reasonable! Will do 👍

**Comment by TorstenDittmann** [comment]
fyi: still working on this, but I am close to my exams right now - so my time is quite limited. Gonna make this jQuery compatible and merge it 👍

**Comment by peyerluk** [comment]
There is no rush. Editable.js is our patient workhorse in the background :)

Good Luck on your exams!

**Comment by marcbachmann** [comment]
oops. that close wasn't on purpose. I deleted the base branch as it didn't have any commits.
image

I can't reopen it here

**Comment by marcbachmann** [comment]
After recreating the branch, the reopen works again

**Comment by marcbachmann** [comment]
I've opened #220, so we can land this

**Comment by marcbachmann** [comment]
@TorstenDittmann are you fine if we continue on there?

**Pull request closed without merging**